js/server/modules/@arangodb/foxx/service.js

```diff
@@ -707,28 +707,22 @@ module.exports =
     }
 
     static get _appPath () {
+      let dirName = internal.db._name();
+      if (!internal.isAllowedDatabaseName(dirName, false)) {
+        dirName = internal.db._id();
+      } 
       return APP_PATH ? (
-        path.join(APP_PATH, '_db', internal.db._name())
-        ) : undefined;
+        path.join(APP_PATH, '_db', dirName)
+      ) : undefined;
+
     }
 
     static get _systemAppPath () {
+      // TODO: should this be STARTUP_PATH here?
```

Comment:
What about this TODO?

Reply:
TO my eyes it looks like STARTUP_PATH would be correct, but I also do not know if that is correct.

Reply:
Alan also suggested that this was probably a copy&paste mistake. So I will adjust it now in this PR. If it doesn't work, we can revert this one change.